### PR TITLE
Add additional logging for construction and HeadObject requests

### PR DESCRIPTION
**Pull request by muddyfish**
## Description

Adds logging at INFO level whenever a Dataset is constructed
Adds logging at DEBUG level whenever a HeadObject request is explicitly made from the s3torchconnector side. 

## Additional context

No breaking changes

## Testing

Added unit tests, ran a small example script and saw this in the output:
```
INFO s3torchconnector.s3map_dataset 2024-02-06 16:06:04,593 s3map_dataset.py:98 - Building S3MapDataset from_prefix
```

--------
By submitting this pull request, I confirm that my contribution is made under the terms of BSD 3-Clause License and I agree to the terms of the [LICENSE](https://github.com/awslabs/s3-connector-for-pytorch/blob/main/LICENSE).

### CODE REVIEW

**Review by IsaevIlya — CHANGES_REQUESTED** [review]
Thank you for improving our log visibility.
I'm wondering, we are providing visibility on to `head_object` calls, why not to do it for `get_object`, `put_object` and `list_objects` requests at the same time?

**Comment by muddyfish** [comment]
> Thank you for improving our log visibility. I'm wondering, we are providing visibility on to `head_object` calls, why not to do it for `get_object`, `put_object` and `list_objects` requests at the same time?

I can do it for `list_objects`, but `get_object` and `put_object` don't match one to one with calls to S3. If you think it's still useful, we can still add them with that knowledge

**Comment by IsaevIlya** [comment]
> I can do it for `list_objects`, but `get_object` and `put_object` don't match one to one with calls to S3. If you think it's still useful, we can still add them with that knowledge
I totally agree on your comment, but I think it would be nice to know when we are explicitly calling these methods, even thought we can't translate them 1-to-1 to requests to S3.

File: s3torchconnector/src/s3torchconnector/s3iterable_dataset.py [hunk 1]
```diff
@@ -83,6 +87,7 @@ def from_prefix(
         Raises:
             S3Exception: An error occurred accessing S3.
         """
+        log.info(f"Building {cls.__name__} from_prefix")
```
Comment (dnanuti) [inline s3torchconnector/src/s3torchconnector/s3iterable_dataset.py:90]: Should we also include the prefix here? 
For objects_uris is too much to log, but prefix might be useful.

**Review by dnanuti — DISMISSED** [review]

**Comment by dnanuti** [comment]
nit: it's failing Python lints

**Review by IsaevIlya — DISMISSED** [review]

**Review by IsaevIlya — APPROVED** [review]

**Review by dnanuti — APPROVED** [review]

**Pull request merged**